### ISSUE TITLE
The relationship with FullCRF

### ISSUE THREAD

**Issue by zy937085318** [MarvinTeichmann/ConvCRF#10]
hi~ I'm digging in your work for weeks and it's really awesome! Now I wanna ask if I set the conv size as the input size, does it mean that the convCRF is equivalent to FullyCRF in this case?

**Comment by hsu-z2** [comment]
My understanding is you need to change the kernel size in the message passing function. Change it from 7 (the value used in the current code) to your input size.

**Comment by MarvinTeichmann** [comment]
Yes, if you use a large filter size you will get a model very similar to FullCRF. Note that there are some subtle differences. FullCRF uses the Permutohedral Lattice approximation, while a ConvCRF with large filter size is exact. 

I have visualized the differences by plotting the impulse response for the filters on a simple 10x10 pixel image, where the center pixels are white with a black frame. The 10x10 filter has the same response as a simple (inefficient) exact filter I have implemented in numpy. The lattice filter however loses some detail. Below the results.

| Input Image: 10 x 10 | Response of a 5x5 filter | Response of a 10x10 filter | Response of Lattice (FullCRF)
| -------- | -------- | -------- | -------- |
| ![input](https://i.imgur.com/kKebLOQ.png)     | ![5filter](https://i.imgur.com/ZGOxmGH.png)     | ![filter10](https://i.imgur.com/3U2aFbw.png)     | ![lattice](https://i.imgur.com/jnoComj.png)     |
